_posts/2024-02-26-cache-refresh.md

```diff
@@ -0,0 +1,166 @@
+author: Manfred Moser
+image: /assets/blog/trino-cache-refresh.png
+excerpt_separator: <!--more-->
+show_pagenav: true
+---
+
+Leon Bambrick once said "There are only two hard things in computer science:
+cache invalidation, naming things, and off-by-one errors" improving the initial
+wisdom from [Phil Karlton](https://www.karlton.org/2017/12/naming-things-hard/).
+Well, in the Trino community we know all about caching, naming, off-by-one
+errors, and improving on original wisdom. With the recent [Trino 439
+release]({{site.url}}/docs/current/release/release-438.html) caching from object
+storage file systems got a refresh. Users of catalogs using the Delta Lake,
+Hive, and Iceberg connectors now get to enjoy performance benefits from the new
+Alluxio-powered file system caching. And all that as a replacement from the now
+removed object storage caching with RubiX.
+
+<!--more-->
+
+## In the past
+
+So how did we get here? A long, long time ago Quoble open sourced a [light
+light-weight data caching framework called
+RubiX](https://github.com/qubole/rubix). The library was integrated into the
+Trino Hive connector and enabled [Hive connector storage
+caching]({{site.url}}/docs/438/connector/hive-caching.html). However, over time
+any open source project without active maintenance becomes stale. And like a
+stale cache, a stale open source project can cause issues, or at a minimum
+becomes outdated and unsuitable for modern use. Staleness and other problems
+meant that RubiX had to go.
+
+## Making progress
+
+Time moved on and Trino now includes powerful connectors for the modern
+lakehouse formats Delta Lake, Hudi, and Iceberg:
+
+<div class="container">
+  <div class="row">
+    <div class="col-sm">
+      <a href="{{site.url}}/docs/current/connector/delta-lake.html" target="_blank">
+        <img src="{{site.url}}/assets/images/logos/delta-lake.png" title="Delta Lake connector">
+      </a>
+    </div>
+    <div class="col-sm">
+      <a href="{{site.url}}/docs/current/connector/hudi.html" target="_blank">
+        <img src="{{site.url}}/assets/images/logos/apache-hudi.png" title="Hudi connector">
+      </a>
+    </div>
+    <div class="col-sm">
+      <a href="{{site.url}}/docs/current/connector/iceberg.html" target="_blank">
+        <img src="{{site.url}}/assets/images/logos/apache-iceberg.png" title="Iceberg connector">
+      </a>
+    </div>
+  </div>
+</div>
+
+Hive is still around, just like HDFS, but we consider them both close to legacy
+status. Yet all four connectors could benefit from caching. Good news came at
+Trino Summit 2022 when Hope Wang and Beinan Wang from
+[Alluxio]({{site.url}}/ecosystem/add-on.html#alluxio) presented about their
+integration with Trino and the Hive connector - [Trino optimization with
+distributed caching on data lake]({% post_url
+2023-07-21-trino-fest-2023-alluxio-recap %}). They mentioned plans to open
+source their implementation and an initial PR was created.
```

Comment:
Move `pull request (PR)` up here, as this is the first use of the term.

Reply:
Done